front/src/pages/companies/Companies.tsx

```diff
@@ -74,7 +67,27 @@ export function Companies() {
       variables: newCompany,
       refetchQueries: [getOperationName(GET_COMPANIES) ?? ''],
     });
-  }
+  }, [insertCompany]);
+
+  useEffect(() => {
+    if (queuedActions.includes('companies/create_company')) {
+      handleAddButtonClick();
+      setQueuedActions(
+        queuedActions.filter((action) => action !== 'companies/create_company'),
```

Comment:
I think we could use an array of objects with ids to have an idempotent data structure, what do you think ? 

Here if we have two create company actions, they would be both filtered out.

Reply:
Maybe an order field would be relevant too, let me know if you think it's relevant here, we can also create an issue to save this for enhancement later.

Reply:
@lucasbordeau I have made the suggested changes and tested it, working fine.

> Maybe an order field would be relevant too, let me know if you think it's relevant here, we can also create an issue to save this for enhancement later.

The current queue structure already ensures that actions are processed in the order they were added, and the specific sequence of actions does not seem to matter in our case. Please let me know your thoughts

Reply:
The new structure is ok